make.sh

```diff
@@ -3,30 +3,48 @@
 # Fail on first exception.
 set -e
 
+PROG=$(basename ${0})
+
 # Clean out the build folder then replace the files we need to keep.
-rm -rf build/*
+[ -d "./build" ] && rm -rf build/* || mkdir build
 cp .gitignore LICENSE README.md build
 
 # Set up the folders we will need.
 mkdir -p build/html
 mkdir -p build/notebooks
 mkdir -p build/solutions
 
+# Change to the local build directory.
 cd build
-cp -rf ../course_content/images images
-cp -rf ../course_content/resources resources
 
-for name in "numpy_intro" "matplotlib_intro" "cartopy_intro" "iris_intro"
+# All directories are relative to the build directory.
+CONTENT_DIR="../course_content"
+IMAGES_DIR="${CONTENT_DIR}/images"
+RESOURCES_DIR="${CONTENT_DIR}/resources"
+NOTEBOOKS_DIR="${CONTENT_DIR}/notebooks"
+
+# Check required source course content directories exist.
+for DIR in "${IMAGES_DIR}" "${RESOURCES_DIR}" "${NOTEBOOKS_DIR}"
+do
+    [ ! -d "${DIR}" ] && echo "${PROG}: Missing directory, \"${DIR}\"" && exit 1
+done
+
+# Copy required resources to the local build directory.
+cp -rf ${IMAGES_DIR} images
+cp -rf ${RESOURCES_DIR} resources
+
+# Enumerate the course content notebooks.
+NOTEBOOKS=$(ls -1 ${NOTEBOOKS_DIR}/*.ipynb)
+
+# Build static (HTML) copies of the course content.
+for NOTEBOOK in ${NOTEBOOKS}
 do
-    #ipython nbconvert --to slides ../../course_content/${name}.ipynb
-    # Build static (html) copies of the course content.
-    jupyter nbconvert --to html ../course_content/notebooks/${name}.ipynb
-    mv ../course_content/notebooks/*.html ./html/
+    jupyter nbconvert --to html ${NOTEBOOK}
+    echo
```

Comment:
What does this do in this case?

Reply:
Simply just breaks up the output so that it's easy to see `nbconvert` doing its thing on each of the notebooks i.e.
```shell
itwl@eld114:> ./make.sh 
[NbConvertApp] Converting notebook ../course_content/notebooks/cartopy_intro.ipynb to html
[NbConvertApp] Writing 263297 bytes to ../course_content/notebooks/cartopy_intro.html

[NbConvertApp] Converting notebook ../course_content/notebooks/dask_intro.ipynb to html
[NbConvertApp] Writing 287351 bytes to ../course_content/notebooks/dask_intro.html

[NbConvertApp] Converting notebook ../course_content/notebooks/iris_intro.ipynb to html
[NbConvertApp] Writing 395800 bytes to ../course_content/notebooks/iris_intro.html

[NbConvertApp] Converting notebook ../course_content/notebooks/matplotlib_intro.ipynb to html
[NbConvertApp] Writing 291794 bytes to ../course_content/notebooks/matplotlib_intro.html

[NbConvertApp] Converting notebook ../course_content/notebooks/numpy_intro.ipynb to html
[NbConvertApp] Writing 335519 bytes to ../course_content/notebooks/numpy_intro.html
```
instead of
```shell
itwl@eld114:> ./make.sh 
[NbConvertApp] Converting notebook ../course_content/notebooks/cartopy_intro.ipynb to html
[NbConvertApp] Writing 263297 bytes to ../course_content/notebooks/cartopy_intro.html
[NbConvertApp] Converting notebook ../course_content/notebooks/dask_intro.ipynb to html
[NbConvertApp] Writing 287351 bytes to ../course_content/notebooks/dask_intro.html
[NbConvertApp] Converting notebook ../course_content/notebooks/iris_intro.ipynb to html
[NbConvertApp] Writing 395800 bytes to ../course_content/notebooks/iris_intro.html
[NbConvertApp] Converting notebook ../course_content/notebooks/matplotlib_intro.ipynb to html
[NbConvertApp] Writing 291794 bytes to ../course_content/notebooks/matplotlib_intro.html
[NbConvertApp] Converting notebook ../course_content/notebooks/numpy_intro.ipynb to html
[NbConvertApp] Writing 335519 bytes to ../course_content/notebooks/numpy_intro.html
```
I personally find it easier to read ... minor, but hey, there you go,

Reply:
Oh, simple as that. I like it 👍